safe-global/safe-wallet-web
# Feat: cached safe overviews

## What it solves
Rewrites `useSafeOverviews` hook using RTK query enabling:
- caching
- more efficient batching of requests
- easier interface to query safe overviews

The goal is to have two hooks:
- One that loads a single Safe overview for any Safe.
- One that loads multiple Safe overviews

Both of them use a shared queue that waits for more requests to come in for 50ms or triggers a requests once 10 requests are in the queue.

Other changes
- Do not fetch Safe overviews for undeployed Safes

## How to test it
While monitoring the outgoing network requests via e.g. developer tools in the browser:
- Connect a wallet and open the sidebar / account list.
- Select and open any Safe
- Now re-open the sidebar. 
- This should not trigger any new requests as the data should still be cached.

## Stats
On a full sidebar with multichain Safes this optimization with a fresh browser state reduces the safe overview requests from 38 to 18.

## Checklist
